### DISCUSSION TITLE
Carousel Component

### DISCUSSION THREAD

**Discussion by ghost** [tailwindlabs/headlessui#1011] [Ideas]
Would be good to have an accessible carousel component.

It should support custom amount of items shown, items to slide, snapping, and should focus onto the new slide that comes into view/ie centred.

**Comment by yuhang1995** [comment]
[embla-carousel](https://github.com/davidcetinkaya/embla-carousel)  very good

**Reply by ghost** [reply]
Using it, it is amazing :D I hope tailwind makes something similar so that we have everything in one package :D

**Reply by aaronadamsCA** [reply]
Unfortunately that library isn't headless and doesn't use DOM scrolling.

I think a more relevant piece of prior art is [react-snap-carousel](https://github.com/richardscarrott/react-snap-carousel) ([Storybook](https://richardscarrott.github.io/react-snap-carousel/)), which simply ships a hook you can use to build your own components. There's a lot of excellent prior art there that could potentially be adapted into a set of Headless UI components.

If anyone knows of any other headless carousel/swiper libraries, please share!

**Comment by seanonthenet** [comment]
+1 Carousel

**Comment by pieer** [comment]
+1 carousel

**Comment by hananint** [comment]
+1 carousel

**Comment by michael-farwell** [comment]
+1 carousel

**Comment by aonoriaga** [comment]
+1 carousel

**Comment by dalkommatt** [comment]
+1 carousel

**Comment by fkunn1326** [comment]
+1 carousel

**Comment by 88Navistar** [comment]
+ 1 carousel '23

**Comment by yxxh020** [comment]
+1 carousel

**Comment by mugnimaestra** [comment]
+1 Carousel

**Comment by ysuliman** [comment]
+1

**Comment by shadiramadan** [comment]
I implemented a carousel using headless UI tabs (fade transition), the problem is that you can't do an exit transition- only an entrance as it appears the non-displayed component is completely removed from the dom.

It would be really nice to support a native headless Carousel that can support all kinds of transitions!

**Comment by Angeschossen** [comment]
+1

**Comment by lesofger** [comment]
+1

**Comment by suuf** [comment]
+1 '26 :')
